pkg/kapp/resources/labeled_resources.go

```diff
@@ -135,6 +143,28 @@ func (a *LabeledResources) AllAndMatching(newResources []Resource, opts AllAndMa
 	return resources, nil
 }
 
+func (a *LabeledResources) resourcesForOwnershipCheck(newResources []Resource, nonLabeledResources []Resource) []Resource {
+	var resources []Resource
+
+	resourcesToBeSkipped := map[string]bool{}
+
+	for _, res := range newResources {
+		_, hasExistsAnnotation := res.Annotations()[ExistsAnnKey]
+		_, hasNoopAnnotation := res.Annotations()[NoopAnnKey]
+		if hasExistsAnnotation || hasNoopAnnotation {
+			resourcesToBeSkipped[NewUniqueResourceKey(res).String()] = true
+		}
+	}
+
+	for _, res := range nonLabeledResources {
+		if !resourcesToBeSkipped[NewUniqueResourceKey(res).String()] {
+			resources = append(resources, res)
+		}
+	}
```

Comment:
(I  have read it a few times, but I am still not sure if I have understood it properly 😅 )

> or is this not how it works today since because we only find non labeled resources based on new resources?

Yeah, we find the non labeled resources based on the new resources and the labeled resources. 
https://github.com/vmware-tanzu/carvel-kapp/blob/31e7abfc48f00ab81fda9ca0235062e9e33ca7c1/pkg/kapp/resources/labeled_resources.go#L213